packages/voting/src/voting-calls.ts

```diff
@@ -877,10 +877,18 @@ export namespace VotingApi {
      * @param filters Optional criteria to filter the processes ID's given by the gateway
      * @param gateway
      */
-    export async function getProcessList(filters: { entityId?: string, namespace?: number, status?: VochainProcessStatus, withResults?: boolean, from?: number } = {}, gateway: IGatewayDVoteClient): Promise<string[]> {
+    export async function getProcessList(filters: { fromArchive: true, entityId?: string } | { fromArchive: false, entityId?: string, namespace?: number, status?: VochainProcessStatus, withResults?: boolean, from?: number }, gateway: IGatewayClient): Promise<string[]>
+    {
         if (!gateway) throw new Error("Invalid gateway client")
         else if (typeof filters != "object") throw new Error("Invalid filters parameter")
 
+        if (filters.fromArchive) {
```

Comment:
Do we want to list all processes from all entities that ever existed?
IMO, the `entityId` field should be mandatory when `fromArchive` is true. At least the other flow will paginate things, but fetching the entire archive can be an easy DOS (that could be done anyway)

Reply:
> Do we want to list all processes from all entities that ever existed?

IMHO yes, the explorer is a use case, for example, but there can be many others

> IMO, the entityId field should be mandatory when fromArchive is true. At least the other flow will paginate things, but fetching the entire archive can be an easy DOS (that could be done anyway)

For me it shouldn't be mandatory... If you can call the GW request without entityId, why here not? DOS isn't possible because of the missing entityId, because no matter what you request, the whole IPFS index file will be requested.